data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/ExportManager.java

```diff
@@ -68,59 +93,146 @@ abstract void processFooter(
   public ExportReport startExport(
       ExportOptions exportOptions, TableMetadata tableMetadata, Writer writer) {
     ExportReport exportReport = new ExportReport();
+    ExecutorService executorService = null;
+
     try {
       validateExportOptions(exportOptions, tableMetadata);
-      Map<String, DataType> dataTypeByColumnName = tableMetadata.getColumnDataTypes();
       handleTransactionMetadata(exportOptions, tableMetadata);
-      processHeader(exportOptions, tableMetadata, writer);
 
-      int maxThreadCount =
-          exportOptions.getMaxThreadCount() == 0
-              ? Runtime.getRuntime().availableProcessors()
-              : exportOptions.getMaxThreadCount();
-      ExecutorService executorService = Executors.newFixedThreadPool(maxThreadCount);
+      try (BufferedWriter bufferedWriter = new BufferedWriter(writer)) {
+        processHeader(exportOptions, tableMetadata, bufferedWriter);
 
-      BufferedWriter bufferedWriter = new BufferedWriter(writer);
-      boolean isJson = exportOptions.getOutputFileFormat() == FileFormat.JSON;
+        int threadCount =
+            exportOptions.getMaxThreadCount() > 0
+                ? exportOptions.getMaxThreadCount()
+                : Runtime.getRuntime().availableProcessors();
+        executorService = Executors.newFixedThreadPool(threadCount);
```

Comment:
My understanding is as follows:

- The main thread (a single thread) scans the table.
- Multiple threads process the scan results in batches and write them through a single writer (a single file).

Is that correct?

Reply:
Yes, that is correct.